templates/buy-me-a-coffee-app/src/features/BuyMeACoffee/components/FormBuyCoffee.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { useState, useCallback } from 'react';
+
+import { useWaitForTransaction, usePrepareContractWrite, useContractWrite } from 'wagmi';
+
+import { parseEther } from 'viem';
+import { Card, Text, Flex, Box, Button, Heading, TextFieldInput, TextArea } from '@radix-ui/themes';
+
+import { contractAddress, contractABI } from '../contract/contractInfo';
+
+type FormBuyCoffeeProps = {
+  onComplete: () => void;
+};
+
+function FormBuyCoffee({ onComplete }: FormBuyCoffeeProps) {
+  // Component state
+  const [name, setName] = useState('');
+  const [message, setMessage] = useState('');
+
+  // Wagmi Write call
+  const { config } = usePrepareContractWrite({
+    address: contractAddress as `0x${string}`,
+    abi: contractABI,
+    functionName: 'buyCoffee',
+    args: [name, message],
+    enabled: name !== '' && message !== '',
+    value: parseEther('0.001'),
+    onSuccess(data) {
+      console.log('Success prepare buyCoffee', data);
+    },
+  });
+
+  // Wagmi Write call
+  const { write: buyMeACoffee, data: dataBuyMeACoffee } = useContractWrite({
+    ...config,
+    onSuccess(data) {
+      console.log('Success write buyCoffee', data);
+    },
+  });
+
+  const { isLoading: loadingTransaction } = useWaitForTransaction({
```

Comment:
Let's take a minute and polish a bit more with documentation on each of these steps how they work, and where to go learn more about it.